grails-app/controllers/UserProfileController.groovy

```diff
@@ -0,0 +1,100 @@
+import org.transmart.searchapp.AuthUser
+import org.transmartproject.db.log.AccessLogService
+
+class UserProfileController {
+
+	def springSecurityService
+	def userProfileService
+	def auth0Service
+	def userService
+	def auth0Config
+	AccessLogService accessLogService
+
+	// non-recoverable error
+	String severeMessage = "Unable to update user information. Contact administrator."
+
+	def index() {
+		try{
+			def userDetails
+			def token
+
+			token = auth0Service.jwtToken()
+			userDetails = userService.currentUserInfo()
+
+            model:[
+				user: userDetails,
+				token: token?: "Unable to retrieve token.",
+				instanceType: auth0Config.instanceType,
+				instanceName: auth0Config.instanceName,
+				level: userDetails.level
+			]
+
+		}
+		catch (Exception e) {
+			log.error("Caught error in UserProfile plugin.", e);
+			redirect(action: 'basic')
+		}
+	}
+
+	def save() {
+
+		try {
+			AuthUser authUser
+			String email     = params.email
```

Comment:
When the local variable name is the same as the param name, you can declare the variable in the method signature and Grails will set the value and convert to the specified type, e.g. `def save(String email, String firstname, String lastname) {`